tests/src/services/users.service.test.ts

```diff
@@ -0,0 +1,235 @@
+      (usersRepository.findUserByUuid as jest.Mock).mockResolvedValue(mocks.mockedUser);
+
+      const result = await usersService.findUserByUuid(mocks.mockedUser.uuid);
+
+      expect(result).toEqual(mocks.mockedUser);
+      expect(usersRepository.findUserByUuid).toHaveBeenCalledTimes(1);
+      expect(usersRepository.findUserByUuid).toHaveBeenCalledWith(mocks.mockedUser.uuid);
+    });
+
+    it('should throw UserNotFoundError when no user is found by UUID', async () => {
+      (usersRepository.findUserByUuid as jest.Mock).mockResolvedValue(null);
+
+      await expect(usersService.findUserByUuid(mocks.mockedUser.uuid)).rejects.toThrow(UserNotFoundError);
+
+      expect(usersRepository.findUserByUuid).toHaveBeenCalledTimes(1);
+      expect(usersRepository.findUserByUuid).toHaveBeenCalledWith(mocks.mockedUser.uuid);
+    });
+  });
+
+  describe('Cancelling user subscription', () => {
+    it('Cancel the user individual subscription', async () => {
+      jest
+        .spyOn(paymentService, 'getActiveSubscriptions')
+        .mockImplementation(() =>
+          Promise.resolve(
+            mocks.mockActiveSubscriptions.filter(
+              (sub) => sub.product?.metadata.type !== 'business',
+            ) as unknown as ExtendedSubscription[],
+          ),
+        );
+      const cancelSubscriptionSpy = jest.spyOn(paymentService, 'cancelSubscription').mockImplementation(voidPromise);
+      const changeStorageSpy = jest.spyOn(storageService, 'changeStorage').mockImplementation(voidPromise);
+
+      await paymentService.getActiveSubscriptions(mocks.mockedUser.customerId);
+
+      await usersService.cancelUserIndividualSubscriptions(mocks.mockedUser.customerId);
+      await storageService.changeStorage(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+
+      const individualSubscriptions = mocks.mockActiveSubscriptions.filter(
+        (sub) => sub.product?.metadata.type !== 'business',
+      );
+      expect(cancelSubscriptionSpy).toHaveBeenCalledTimes(individualSubscriptions.length);
+
+      expect(changeStorageSpy).toHaveBeenCalledTimes(1);
+      expect(changeStorageSpy).toHaveBeenCalledWith(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+    });
+
+    it('Cancel the user B2B subscription', async () => {
+      jest
+        .spyOn(paymentService, 'getActiveSubscriptions')
+        .mockImplementation(() => Promise.resolve(mocks.mockActiveSubscriptions as unknown as ExtendedSubscription[]));
+
+      const cancelSubscriptionSpy = jest.spyOn(paymentService, 'cancelSubscription').mockImplementation(voidPromise);
+
+      const changeStorageSpy = jest.spyOn(storageService, 'changeStorage').mockImplementation(voidPromise);
+
+      await usersService.cancelUserB2BSuscriptions(mocks.mockedUser.customerId);
+      await storageService.changeStorage(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+
+      const b2bSubscriptions = mocks.mockActiveSubscriptions.filter((sub) => sub.product?.metadata.type === 'business');
```

Comment:
It's better to separate between business and individual subscriptions in the mocks rather than using logic (like filtering) in a test, as any kind of logic should be tested.